### PR TITLE
Feat/interest txn history

### PR DESCRIPTION
**Pull request by jjBlockchain**
## Description (optional)
* Add Interest transaction history download button to convert selected coin's interest transactions into a csv file.
* **Not Asked:**
Screen Shot 2021-04-23 at 12 18 19 AM

* **Loading:**
Screen Shot 2021-04-23 at 12 24 17 AM

* **Success:**
Screen Shot 2021-04-23 at 12 18 28 AM

* **Success after saving csv:**
Screen Shot 2021-04-23 at 12 18 35 AM

* **Failed:**
Screen Shot 2021-04-23 at 12 19 16 AM

## Testing Steps (optional)
* Login into your wallet account with interest transactions
* Click Earn Interest on the top navigation
* Click Transaction History tab
* After initial transactions load, download button should show
* Click download
* Wait for data to finish fetching
* Click Save Report
* Should see button change to Saved

### CODE REVIEW

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 1]
```diff
@@ -113,7 +113,7 @@ export default ({
     }
   }
 
-  const fetchInterestLimits = function * ({
+  const fetchInterestLimits = function*({
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:116]: why are all these formatting changed?  Seems like a lint rule difference, update your branch with latest `development` and address

File: packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts [hunk 2]
```diff
@@ -159,7 +159,41 @@ export default ({
...
+  const fetchInterestTransactionsReport = function*() {
+    const reportHeaders = [['Date', 'Type', 'Asset', 'Amount', 'Tx Hash']]
+    const formatTxData = d => [
+      d.insertedAt,
+      d.type,
+      d.amount?.symbol,
+      d.amount?.value,
+      d.txHash
+    ]
+    let list = [],
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/data/components/interest/sagas.ts:171]: `txList`

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+  text-decoration: none;
+`
+const StyledIconButton = styled(IconButton)`
+  ${IconButtonCss};
+  color: ${props => props.theme['blue600']};
+`
+const SuccessIconButton = styled(IconButton)`
+  ${IconButtonCss};
+  color: ${props => props.theme.white};
+  background-color: ${props => props.theme.success};
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:29]: can we just use the color codes directly? I think this should be `blue600`

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+  opacity: 1;
+`
+const FailedIconButton = styled(IconButton)`
+  ${IconButtonCss};
+  color: ${props => props.theme.white};
+  background-color: ${props => props.theme.error};
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:35]: `red600`

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+import { bindActionCreators, Dispatch } from 'redux'
+import styled, { css } from 'styled-components'
+
+import { HeartbeatLoader, IconButton } from 'blockchain-info-components'
+import { actions, selectors } from 'data'
+import { InterestHistoryCoinFormType } from 'data/components/interest/types'
+import { RootState } from 'data/rootReducer'
+
+const IconButtonCss = css`
+  border: 1px solid ${props => props.theme['grey100']};
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:15]: `props.theme.grey100`

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+  border-radius: 8px;
+  margin-right: 12px;
+`
+const DownloadButton = styled(CSVLink)`
+  text-decoration: none;
+`
+const StyledIconButton = styled(IconButton)`
+  ${IconButtonCss};
+  color: ${props => props.theme['blue600']};
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:24]: `props.theme.blue600`

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+            data-e2e='generateInterestTxReport'
+            height='45px'
+            onClick={this.handleDownload}
+            name='download'
+            nature='light'
+            width='140px'
+          >
+            <FormattedMessage
+              id='scenes.transactions.export.download'
+              defaultMessage='Download'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:147]: `copy.download`?

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+              data-e2e='interestTxReportSaved'
+              disabled
+              height='45px'
+              name='checkmark-in-circle-filled'
+              nature='primary'
+              width='140px'
+            >
+              <FormattedMessage
+                id='scenes.interest.transactions.saved'
+                defaultMessage='Saved'
```
Comment (schnogz) [inline packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:94]: `button.saved`?

**Review by schnogz — CHANGES_REQUESTED** [review]
Looks pretty good, mostly small knitpicks.

**Comment by pedroapfilho** [comment]
Being very picky, but could you create the loading variation as a button and not as an icon? It would make it look nicer

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx (hunk 7 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:147]: There is not a copy.download, should I change all the other Downloads to copy.download?

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx (hunk 3 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:29]: The success color is unique (#00875A), should I use green600 (#00994C)

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx (hunk 4 above)
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:35]: Same here with red600(#CF1726), it does not match error(#D93B30)

File: packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+              />
+            </SuccessIconButton>
+          ) : (
+            <DownloadButton
+              data={val}
+              filename='Interest_Transactions.csv'
+              onClick={this.handleSaveReport}
+              target='_blank'
```
[the remark this answers is not in the document]
Comment (jjBlockchain) [reply packages/blockchain-wallet-v4-frontend/src/scenes/InterestHistory/DownloadTransactions/index.tsx:102]: What do you mean by autoclose?

**Review by pedroapfilho — APPROVED** [review]

**Review by schnogz — APPROVED** [review]

**Pull request merged**